src/hotspot/os/linux/cgroupV1Subsystem_linux.cpp

```diff
@@ -241,3 +241,30 @@ int CgroupV1Subsystem::cpu_shares() {
 
   return shares;
 }
+
+/* pids_max
+ *
+ * Return the maximum number of tasks available to the process
+ *
+ * return:
+ *    maximum number of tasks
+ *    -1 for no setup
+ *    -3 for "max" (special value)
+ *    OSCONTAINER_ERROR for not supported
+ */
+jlong CgroupV1Subsystem::pids_max() {
+  if (_pids == NULL) return OSCONTAINER_ERROR;
+
+  GET_CONTAINER_INFO(jlong, _pids, "/pids.max",
+                     "Maximum number of tasks is: " JLONG_FORMAT, JLONG_FORMAT, pidsmax);
+  if (pidsmax < 0) {
+    // check for potential special value
+    char myline[1024];
+    int err2;
+    err2 = subsystem_file_line_contents(_pids, "/pids.max", NULL, "%1023s", myline);
+    if (err2 != 0) {
+      if (strncmp(myline, "max", 3) == 0) return -3;
```

Comment:
This looks like it should use `GET_CONTAINER_INFO_CPTR` macro and then `limit_from_str` from cgroups v2 code. Perhaps move `limit_from_str` method to the base class.